drivers/interrupt_controller/intc_loapic.c

```diff
@@ -59,18 +62,15 @@
 #define LOAPIC_SPURIOUS_VECTOR_ID CONFIG_LOAPIC_SPURIOUS_VECTOR_ID
 #endif
 
-#define LOPIC_SSPND_BITS_PER_IRQ  1  /* Just the one for enable disable*/
-#define LOPIC_SUSPEND_BITS_REQD (ROUND_UP((LOAPIC_IRQ_COUNT * LOPIC_SSPND_BITS_PER_IRQ), 32))
+#define LOAPIC_SSPND_BITS_PER_IRQ  1  /* Just the one for enable disable*/
+#define LOAPIC_SUSPEND_BITS_REQD (ROUND_UP((LOAPIC_IRQ_COUNT * LOAPIC_SSPND_BITS_PER_IRQ), 32))
```

Comment:
Do you still use hardcoded `LOAPIC_IRQ_COUNT` ? Is this the same  `num-irqs` from DTS?

Reply:
This should be the same for `num-irqs`. However, in this PR, I have not changed much of interrupt controller driver code. For instance, a hardcoded number of 0x320 is used for timer interrupt.I recognize at least `IRQ_COUNT` can be replaced with `num-irqs`. However, currently irq mapping is done in static way in the interrupt controller code itself (https://github.com/zephyrproject-rtos/zephyr/blob/ef5d0fd291568e39d55ef95792e217146d6c2043/drivers/interrupt_controller/intc_loapic.c#L215). I am not sure how much of this should be covered in the scope of current PR. Perhaps, another run on the interrupt controller code would be appropriate.

Reply:
Can we do some macrobatic and get `LOAPIC_IRQ_COUNT` from that `num-irq`, probably need to rename `IRQ_CONT` to `NUM_IRQ`.

Reply:
Probably we can do something like `DT_NUM_IRQS(DT_NODELABEL(intc_loapic))` where `LOAPIC_IRQ_COUNT` is defined. However for now I will revert the `num-irqs` change as I think this change could be addressed along with the other potential changes in interrupt controller to get rid of various other hard-coded values and source it from the description of the node.